src/frontend/hooks/useWhyDidYouUpdate.ts

```diff
@@ -0,0 +1,35 @@
+import { useRef, useEffect } from 'react'
+
+export function useWhyDidYouUpdate(
```

Comment:
for debugging purposes only.
Can be used to check props, state and context changes on a functional component.
It logs all changes on the passed variables.
got from:
https://usehooks.com/useWhyDidYouUpdate/